streams/src/test/java/org/apache/kafka/streams/tests/EosTestDriver.java

```diff
@@ -559,47 +601,36 @@ private static void verifyAllTransactionFinished(final KafkaConsumer<byte[], byt
                         exception.printStackTrace(System.err);
                         System.err.flush();
                         Exit.exit(1);
+                    } else {
+                        endMarkerOffset.put(new TopicPartition(metadata.topic(), metadata.partition()), metadata.offset());
+                        System.out.println("Appended verification record to topic-partition " + metadata.topic() + "-" + metadata.partition() + " at offset " + metadata.offset());
+                        System.out.flush();
                     }
                 });
             }
         }
 
-        final StringDeserializer stringDeserializer = new StringDeserializer();
-
         long maxWaitTime = System.currentTimeMillis() + MAX_IDLE_TIME_MS;
-        while (!partitions.isEmpty() && System.currentTimeMillis() < maxWaitTime) {
-            final ConsumerRecords<byte[], byte[]> records = consumer.poll(Duration.ofMillis(100));
-            if (records.isEmpty()) {
-                System.out.println("No data received.");
-                for (final TopicPartition tp : partitions) {
-                    System.out.println(tp + " at position " + consumer.position(tp));
-                }
-            }
-            for (final ConsumerRecord<byte[], byte[]> record : records) {
-                maxWaitTime = System.currentTimeMillis() + MAX_IDLE_TIME_MS;
-                final String topic = record.topic();
-                final TopicPartition tp = new TopicPartition(topic, record.partition());
+        while (!endMarkerOffset.isEmpty() && System.currentTimeMillis() < maxWaitTime) {
+            consumer.seekToEnd(partitions);
```

Comment:
`seekToEnd()` will only reach the end-marker in read-committed mode, if there is no pending transaction.